src/Transport/Configuration/AzureServiceBusTransportSettingsExtensions.cs

```diff
@@ -92,55 +92,53 @@ public static TransportExtensions<AzureServiceBusTransport> TimeToWaitBeforeTrig
-        /// <param name="transportExtensions"></param>
-        /// <param name="subscriptionNameShortener">The callback to apply.</param>
-        /// <returns></returns>
-        public static TransportExtensions<AzureServiceBusTransport> SubscriptionNameShortener(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> subscriptionNameShortener)
+        /// <param name="transportExtensions">The transport settings object</param>
+        /// <param name="subscriptionNameConvention">The callback to apply.</param>
+        public static TransportExtensions<AzureServiceBusTransport> SubscriptionNameConvention(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> subscriptionNameConvention)
         {
-            Guard.AgainstNull(nameof(subscriptionNameShortener), subscriptionNameShortener);
+            Guard.AgainstNull(nameof(subscriptionNameConvention), subscriptionNameConvention);
 
-            Func<string, string> wrappedSubscriptionNameShortener = subsciptionName =>
+            Func<string, string> wrappedSubscriptionNameConvention = subsciptionName =>
             {
                 try
                 {
-                    return subscriptionNameShortener(subsciptionName);
+                    return subscriptionNameConvention(subsciptionName);
                 }
                 catch (Exception exception)
                 {
-                    throw new Exception("Custom subscription name shortener threw an exception.", exception);
+                    throw new Exception("Custom subscription name convention threw an exception.", exception);
                 }
             };
 
-            transportExtensions.GetSettings().Set(SettingsKeys.SubscriptionNameShortener, wrappedSubscriptionNameShortener);
+            transportExtensions.GetSettings().Set(SettingsKeys.SubscriptionNameConvention, wrappedSubscriptionNameConvention);
 
             return transportExtensions;
         }
 
         /// <summary>
-        /// Specifies a callback to apply to a subscription rule name when a subscribed event's name is longer than 50 characters.
+        /// Specifies a callback to apply to customize a subscription rule name.
         /// </summary>
-        /// <param name="transportExtensions"></param>
-        /// <param name="ruleNameShortener">The callback to apply.</param>
-        /// <returns></returns>
-        public static TransportExtensions<AzureServiceBusTransport> RuleNameShortener(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> ruleNameShortener)
+        /// <param name="transportExtensions">The transport settings object</param>
+        /// <param name="ruleNameConvention">The callback to apply.</param>
+        public static TransportExtensions<AzureServiceBusTransport> RuleNameConvention(this TransportExtensions<AzureServiceBusTransport> transportExtensions, Func<string, string> ruleNameConvention)
```

Comment:
Similar, this should be "RuleNamingConvention", but now that seems too generic. Should this actually be "SubscriptionRuleNamingConvention" ?

Reply:
`SubscriptionRuleNamingConvention` is a bit chatty but logically a rule is always part of a subscription. We're are replacing the `RuleNameShortener()` API but this could be a good change.